### ISSUE TITLE
Large Scale Point Cloud Classification

### ISSUE THREAD

**Issue by jlevy44** [rusty1s/pytorch_geometric#569]
This repo seems to be interesting: https://github.com/loicland/superpoint_graph

Suppose I have a point set where each training sample is 100k samples by 10k dimensional embeddings. Would it be possible to classify the entire point cloud, each point has 10k dimensions, or computationally intractable?

**Comment by rusty1s** [comment]
Hi, although 100k points with 10k dimensions seem quite high, this all depends on your architecture and how many edges your resulting graph has. How does the linked repo/papers tackle this problem?

**Comment by jlevy44** [comment]
It’s an unrelated question. I’m just curious about fast large n-d point cloud classification. Imagine if 100k to 1 million points of 10k dimensions constituted a point cloud. Having 200 point clouds with labels. No adjacency constructed yet. What would be your recommended architecture to tackle the problem?

**Comment by jlevy44** [comment]
It’d be tough enough to fit the data on the gpu if the gpu is 12g. Some reduction might be warranted. I wonder if you could build the adjacency first outside of gpu with knn and then fps? Then push the data onto gpu? But maybe that will not be efficient?

**Comment by rusty1s** [comment]
The paper linked above mentions the sampling of sampling of points from your initial point cloud, so that might be an option to reduce the number of points in your point cloud. `knn` can now be efficiently computed on the GPU with the [keOps library](https://www.kernel-operations.io/keops/index.html). To reduce the number of dimensions, you may want to apply some dimensionality reduction techniques.

**Comment by jlevy44** [comment]
Thanks! I'll let you know if I have more Qs.
